### ISSUE TITLE
Make meshnets packageble 

### ISSUE THREAD

**Issue by AugustoPeres** [inductiva/meshnets#44]
At the moment there is no way to install meshnets as package so that it can be used outside the main `/meshnets` folder. Ideally we would have this type of folder structure:

```
.
├── # This is the front page of the repo meshnets
├── setup.py
└── meshnets/
    ├── __init__.py
    ├── modules/
    │   ├── decoder.py
    │   └── __init__.py
    ├── datatypes
    └── utils
```

Where the setup file will be something like:

```python
"""Setup script"""
from setuptools import setup, find_packages

setup(name='meshenets', version='1.0', packages=find_packages())
```

And the `__init__.py` would be something like:

```python
# meshenets/modules/__init__.py
from .decoder import *
```

This would allow users to install meshenets by:

1. Cloning the repository;
2. `cd` into the repository;
3. Running `pip install -r requirements.txt`;
4. Running `pip install -e .`

Now meshnets can even be used outside of the repo by simply using:

```python
# some/other/part/of/my/computer/example.py
import meshnets
```

Addressing this issue will consist on creating the `setup.py` file and the `__init__.py_files`. This would even allow to move the python scripts such as [`meshnets/train.py`](https://github.com/inductiva/meshnets/blob/main/meshnets/train.py) to a folder `scripts/train.py`

**Comment by IvanPombo** [comment]
This seems good! Should we also discuss the other ml package where some of the code here is coming from? Or are you thinking of this being a completely independent one?

**Comment by AugustoPeres** [comment]
> This seems good! Should we also discuss the other ml package where some of the code here is coming from? Or are you thinking of this being a completely independent one?

The other ML packages are already implemented like this. I think that mesh nets should be separate.

**Comment by sarmento** [comment]
Thank you for this. I am really happy that we are moving in the direction of having high quality open-source contributions. @hpenedones : does it make sense to have a blog post about this?

**Comment by hpenedones** [comment]
> Thank you for this. I am really happy that we are moving in the direction of having high quality open-source contributions. @hpenedones : does it make sense to have a blog post about this?

Yes, I think it does make sense to have a blog post. Victor will be less available in the coming weeks, but we can work on it among the rest of us.
